include/nuttx/spinlock.h

```diff
@@ -51,6 +52,21 @@ typedef uint8_t spinlock_t;
 
 #include <arch/spinlock.h>
 
+struct fair_spinlock_s
```

Comment:
we shouldn't add new fair_spinlock_s, but:

1. Improve spinlock_t directly
2. Add Kcoonfig to enable it

Reply:
Sorry I can’t got it.
So we don’t need this lock?

Reply:
Ok. Thanks your reply.